### ISSUE TITLE
Boost: Show Performance Improvement to Users

### ISSUE THREAD

**Issue by davidlonjon** [Automattic/jetpack#20680]
The high-level definition for this task can be found on the Project Thread: Show Performance Improvement to Users P2 post: pbIAo1-hO-p2

The first design iteration can be found on the Performance Improvement & Rating i1 post: pbIAo1-iI-p2

Most of the work related to this new feature should be contained within the following areas:

On the PHP side:
- https://github.com/Automattic/jetpack/blob/master/projects/plugins/boost/app/lib/class-speed-score.php
- https://github.com/Automattic/jetpack/blob/master/projects/plugins/boost/app/lib/class-speed-score-request.php

On the JavaScript side: 
- https://github.com/Automattic/jetpack/blob/master/projects/plugins/boost/app/assets/src/js/pages/settings/sections/Score.svelte
- https://github.com/Automattic/jetpack/blob/master/projects/plugins/boost/app/assets/src/js/api/speed-scores.ts
- https://github.com/Automattic/jetpack/blob/master/projects/plugins/boost/app/assets/src/js/pages/settings/elements/ScoreBar.svelte
- https://github.com/Automattic/jetpack/blob/master/projects/plugins/boost/app/assets/src/css/components/score.scss

In a nutshell (open for discussions and of course improvements too) the following high level tasks would need to be done:
- Refactor the way that the site speed score is stored. Maybe have a `previous` and `current` root level key in the scores storage. Make the previous key holding the current scores when a new score comes in. Add some logic to make sure that if (at least one) of previous scores (mobile/desktop) is less than the related new current scores then either do not show the previous score UI (on the JS side) or do not stored it all together (on the PHP side).
- Refactor the UI on the Site Score bar to display the previous (if any) and current site scores.
- Add a new UI component for the plugin rating if there is a positive increase in the score.
